src/Sentry.Unity/NativeUtils/SentryPlatformServices.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+
+namespace Sentry.Unity.NativeUtils;
+
+/// <summary>
+/// These are SDK's services that are only available at runtime and cannot be baked into the SDK. The
+/// <c>SentryInitialization.cs</c> is provided as <c>.cs</c> and gets compiled with the game. It sets <c>IUnityInfo</c>
+/// and the <c>PlatformConfiguration</c> callback during the game's startup so that they are available during initializtion.
+/// </summary>
+/// <remarks>Consider this <c>internal</c>.</remarks>
+public static class SentryPlatformServices
+{
+    private static ISentryUnityInfo? _unityInfo;
+
+    /// <summary>
+    /// The UnityInfo holds methods that rely on conditionally compilation, i.e. IL2CPP backend.
+    /// </summary>
+    public static ISentryUnityInfo UnityInfo
```

Comment:
Example, to try to help the user understand wha'ts going on:


```suggestion
	/// <exception cref="System.InvalidOperationException">
	/// Thrown when unity info was not set since it's expected to be set automatically during Unity initialization.
	/// A common case is when runtime init is disabled, and assignment wasn't done explicitly by the user
	/// </exception>
    public static ISentryUnityInfo UnityInfo
```